cfme/cloud/tenant.py

```diff
@@ -68,10 +72,9 @@ def wait_for_disappear(self, timeout=300):
         except TimedOutError:
             logger.error('Timed out waiting for tenant to disappear, continuing')
 
-    # def wait_for(self, timeout=600):
-    #     return wait_for(self.exists, fail_condition=False, timeout=timeout,
-    #                     message='Wait for cloud tenant to appear', delay=10,
-    #                     fail_func=sel.refresh)
+    def wait_for_appear(self, timeout=600):
+        return wait_for(self.exists, timeout=timeout, message='Wait for cloud tenant to appear',
+                        delay=10, fail_func=sel.refresh)
```

Comment:
I understood you, and it must work as you said, but it doesn't. I don't know the reason, but what I mentioned - it doesn't refresh the page within self.exists().
Here is the part of [cfme.log](http://pastebin.test.redhat.com/486907), started from wait_for_appear call, without fail_func param. As I understand, ressetter() for Tenant/Details was not called.